lib/auth/webauthn/register.go

```diff
@@ -228,20 +228,34 @@ func upsertOrGetWebID(ctx context.Context, user string, identity RegistrationIde
 	return wla.UserID, nil
 }
 
+// RegisterRequest represents fields needed to register a new webautn device.
+type RegistrationRequest struct {
+	// User is the user of the new device.
+	User string
+	// DeviceName is the name for the new device.
+	DeviceName string
+	// CreationResp is the response from the new device.
+	CreationResp *CredentialCreationResponse
+	// Passwordless explicitly requests to register a passwordless device.
+	// This flag will be used to verify that the device response is
+	// passwordless capable.
+	Passwordless bool
```

Comment:
```suggestion
	// Passwordless is true if this is expected to be a passwordless registration.
	// Callers may make certain concessions when processing passwordless
	// registration (such as skipping password validation), this flag reflects that.
	// The data stored in the Begin SessionData must match the passwordless flag,
	// otherwise the registration is denied.
	Passwordless bool
```